python/export_util.py

```diff
@@ -128,6 +129,238 @@ def convert_to_isoformat(
+        data (str): A stream of query results in a cypher format.
+    Raises:
+        PermissionError: If you provided file path that you have no permissions to write at.
+        OSError: If the file can't be opened or written to.
+    """
+
+    cypher = []
+
+    memgraph = gqlalchemy.Memgraph()
+
+    if config.get("write_triggers", True):
+        triggers = memgraph.execute_and_fetch("SHOW TRIGGERS;")
+        for trigger in triggers:
+            cypher.append(
+                f"CREATE TRIGGER {trigger['trigger name']} ON {trigger['event type']} {trigger['phase']} EXECUTE {trigger['statement']};"
+            )
+        cypher.append("")
+
+    if config.get("write_indexes", True):
+        constraints = memgraph.execute_and_fetch("SHOW CONSTRAINT INFO;")
+        for constraint in constraints:
+            constraint_type = constraint["constraint type"]
+
+            if constraint_type == "exists":
+                cypher.append(
+                    f"CREATE CONSTRAINT ON (n:{constraint['label']}) ASSERT EXISTS (n.{constraint['properties']});"
+                )
+            elif constraint_type == "unique":
+                properties = (
+                    [constraint["properties"]]
+                    if isinstance(constraint["properties"], str)
+                    else list(constraint["properties"])
+                )
+                cypher.append(
+                    f"CREATE CONSTRAINT ON (n:{constraint['label']}) ASSERT {'n.' + ', n.'.join(properties)} IS UNIQUE;"
+                )
+            else:
+                raise ValueError("Unknown constraint type.")
+        cypher.append("")
+
+    if config.get("write_constraints", True):
+        indexes = memgraph.execute_and_fetch("SHOW INDEX INFO;")
+        for index in indexes:
+            index_type = index["index type"]
+            if index_type == "label":
+                cypher.append(f"CREATE INDEX ON :{index['label']};")
+            elif index_type == "label+property":
+                cypher.append(
+                    f"CREATE INDEX ON :{index['label']}({index['property']});"
+                )
+            else:
+                raise ValueError("Unknown index type.")
+        cypher.append("")
+
+    graph = get_graph_for_cypher(ctx, config.get("write_properties", True))
+
+    for object in graph:
+        if isinstance(object, Node):
+            object.labels.append("_IMPORT_ID")
+            object.properties["_IMPORT_ID"] = object.id
+            properties_str = (
+                "{"
+                + ", ".join([f"{k}: {v}" for k, v in object.properties.items()])
+                + "}"
+            )
+            cypher.append(f"CREATE (n:{':'.join(object.labels)} {properties_str});")
+        elif isinstance(object, Relationship):
+            properties_str = (
+                "{"
+                + ", ".join([f"{k}: {v}" for k, v in object.properties.items()])
+                + "}"
+            )
```

Comment:
This can be part of function, you can pass properties and it returns joined string